W-14038153 usage reports (#206)
mulesoft/docs-general: modules/ROOT/pages/usage-metrics.adoc

= Usage and Pricing Metrics Reference
:page-aliases: pricing-metrics.adoc

Anypoint Platform captures usage metrics across products, but not all metrics are billable or reflected in usage reports.

[cols=3*]
|===
|Product |Metric Description |Usage Details

.3+|Mule Runtime	

> |_Mule flow_: A flow within a deployed and running Mule app that contains a Mule event source or route APIKit request. 

Reviewer: Why italics here?

Reviewer: Can you just say "running"? Do you need "deployed"?

Reviewer 2: It's that apps can be deployed but might not be running just because they are deployed.

|Flows are aggregated using a Max Concurrent model. The usage for a month is the highest number of flows that exist in a single given hour during a month.

|_Mule message_: The container of the core information processed by the runtime. 
|A Mule message counts as a single unit when an event source triggers it. Messages are aggregated using a sum of all messages sent during a month.
|_Data throughput_: The total amount of data transferred in and out of the infrastructure that runs Mule where the Mule app is deployed. 
|Data throughput counts when the deployed application transfers data to execute its business logic, including but not limited to internal operational network traffic for monitoring, logs, and health checks. Data throughput is aggregated as a sum of all bytes during a month. 

|API Manager	
|_API instance under management_: API instances that are managed by API Manager after they are created using add, promote, or import options. 
|API instances remain under management until they are deleted. Instances of API Manager are aggregated using a Max Concurrent model. 

|API Governance	

> |_API under Governance_: APIs that are identified by the selection criteria of at least one of the governance profiles. 

Reviewer: ```suggestion
|_API under Governance_: APIs identified by the selection criteria of at least one of the governance profiles. 
```

a|If an API is governed, all versions of that API are considered one governed API. Instances of API Governance are aggregated using a Max Concurrent model. The usage for a month is the highest number of APIs governed in a single given hour during a month.

|Flex Gateway	
|_Flex Gateway API call_: Any access request received by Anypoint Flex Gateway regardless of whether the response to the request is successful or not. 
|Flex Gateway requests are aggregated as a sum of all requests during a month.

|Composer	
|_Composer task_: Any action executed on a Composer connector, including but not limited to read, create, update, and delete. 
|Composer tasks are aggregated as a sum of all actions during a month.

|RPA	

> |_Robotic Process Automation (“RPA”) bot minutes_: The number of minutes running process automations across all bot sessions. 

Reviewer: Read this sentence aloud.

Reviewer 2: I did. And the PM says it makes sense.

|A single bot can run multiple parallel sessions, with RPA bot minutes counting for each parallel session. You can configure multiple bots to run the same process, with RPA bot minutes counting for each of these separate bot sessions. Test runs or process runs in the test phase are also counted towards RPA bot minutes.

|Message Queue	
|_Anypoint MQ API request_: A request made to retrieve one or more messages from the Anypoint MQ APIs. 

> |Each Anypoint MQ API request includes up to 100 kilobytes of data. An Anypoint MQ API request over 100 kilobytes counts as multiple requests with no fractional units. Anypoint MQ API requests are calculated in the aggregate across all environments (including production, pre-production, sandbox, and design). MQ API requests are currently available via API and are not currently aggregated on usage reports.

Reviewer: ```suggestion
|Each Anypoint MQ API request includes up to 100 KB of data. An Anypoint MQ API request over 100 KB counts as multiple requests with no fractional units. Anypoint MQ API requests are calculated in the aggregate across all environments (including production, pre-production, sandbox, and design). Anypoint MQ API requests are currently available only via API and aren’t aggregated on usage reports.
```

|Object Store	

> |_Object Store API request_: A request made to retrieve one or more messages from the Object Store APIs as further defined in the Object Store Documentation. 

Reviewer: ```suggestion
|_Object Store API request_: A request made to retrieve one or more messages from the Object Store APIs as further defined in the Object Store documentation. 
```

> |Each Object Store API Request includes up to 100 kilobytes of data. Object Store API Requests over 100 kilobytes count as multiple requests with no fractional units. OS API requests are currently available via API and are not currently aggregated on usage reports.

Reviewer: ```suggestion
|Each Object Store API request includes up to 100 KB of data. Object Store API requests over 100 KB count as multiple requests with no fractional units. Object Store API requests are currently available only via API and aren’t aggregated on usage reports.
```

|DataGraph	
|_DataGraph orchestration_: An API request made by Anypoint DataGraph to the source APIs to get data for the GraphQL API request made to Anypoint DataGraph. 
|Orchestrations are not currently aggregated on usage reports.

| Intelligent Document Processing (IDP)
| Intelligent Document Processing (IDP) Document Pages: A single page processed by IDP.
| IDP document actions might process documents that have more than one page, with each page counting separately. When RPA executes document actions, it also counts towards document pages and additionally consumes the corresponding RPA Bot Minutes, accounting for the time the RPA process runs.
|===

== API Experience Hub Usage

> API Experience Hub usage reports show the following information in cards and tables:

Reviewer: tables and cards (to match the order)?

include::partial$api-eh-metrics.adoc[tag=table]

include::partial$api-eh-metrics.adoc[tag=card]

== API Governance Usage

API Governance usage reports track the maximum number of xref:api-governance::index.adoc#governed-apis[APIs governed] in a given month per business group. 

To calculate usage, MuleSoft tracks the number of governed APIs each business group uses per hour and then identifies the maximum number of APIs governed in a month for each business group. To calculate the total count for API Governance, MuleSoft adds the maximum number of governed APIs per month used across all business groups. 

> The number of governed APIs (both conformant and non-conformant) shown in the API governance console represents a current snapshot of your usage per business group, while the usage report shows the maximum number of APIs you used in a given month per business group.

Reviewer: ```suggestion
The number of governed APIs (both conformant and non-conformant) shown in the API Governance console represents a current snapshot of your usage per business group. The usage report shows the maximum number of APIs you used in a given month per business group.
```

> API governance usage reports show the following information in cards and tables:

Reviewer: ```suggestion
API Governance usage reports show this information in cards and tables:
```

> include::partial$api-governance-metrics.adoc[tag=table]

Reviewer: Where's the card?

== API Manager Usage

API Manager usage reports track the maximum number of xref:api-manager::latest-overview-concept.adoc#api-instances[API instances] deployed in a given month per business group, environment, and runtime. 

To calculate usage, MuleSoft tracks the number of managed API instances each business group, environment, and runtime uses per hour and identifies the maximum number of managed API instances used in a month for each business group, environment, and runtime. To calculate the total count for API Manager, MuleSoft adds the maximum number of managed API instances per month used across business groups, environments, and runtimes.

> API Manager usage reports show the following information in cards and tables:

Reviewer: ```suggestion
API Manager usage reports show this information in cards and tables:
```

> include::partial$apim-metrics.adoc[tag=table]

Reviewer: Where's the card?

== Flex Gateway Usage

Flex Gateway usage reports track daily and monthly API calls. You can view API calls by business group, environment, and registration.

To calculate usage, MuleSoft meters the hourly API calls you make and aggregates the number of API calls on a daily and monthly basis.

> Flex Gateway usage reports show the following information in cards and tables:

Reviewer: ```suggestion
Flex Gateway usage reports show this information in cards and tables:
```

include::partial$flex-gateway-metrics.adoc[tag=card]

include::partial$flex-gateway-metrics.adoc[tag=table]

== Intelligent Document Processing (IDP)

IDP usage reports show the following information in cards and tables:

Business Group::
Business group in which the document is processed.

Action ID::
ID associated with the processed action.

Action Version::
Version associated with the processed action.

Execution Type::
Execution types associated with the processed action.

Processed Pages::
Sum of pages processed.

== Mule Runtime Usage

> Mule Runtime usage reports show the following information in cards and tables:

Reviewer: ```suggestion
Mule runtime usage reports show the following information in cards and tables:
```
